lua/telescope/builtin/lsp.lua

```diff
@@ -182,6 +185,27 @@ lsp.workspace_symbols = function(opts)
   }):find()
 end
 
+lsp.diagnostics = function(opts)
+    local locations = utils.diagnostics_to_tbl(opts)
+    local filename = vim.api.nvim_buf_get_name(0)
+    for _, value in pairs(locations) do value.filename = filename end
```

Comment:
I think best to use `bufnr` over `name` here -- since we already have the buffer.

Reply:
You're right! Info was redundant anyways as per corresponding `make_entry`.